File: ios/Sources/MqttQuicPlugin/QUIC/QuicTypes.swift
//
// QuicTypes.swift
// MqttQuicPlugin
//
// QUIC transport types. Phase 2: ngtcp2 client + single bidirectional stream.
//

import Foundation

/// Represents a single QUIC bidirectional stream (one per MQTT connection).
public protocol QuicStreamProtocol: AnyObject {
    var streamId: UInt64 { get }
    func read(maxBytes: Int) async throws -> Data
    func write(_ data: Data) async throws
    func close() async throws
}

/// QUIC client: connect, TLS handshake, open one bidirectional stream.
public protocol QuicClientProtocol: AnyObject {
    func connect(host: String, port: UInt16) async throws
    func openStream() async throws -> QuicStreamProtocol
    func close() async throws
}

File: android/src/main/kotlin/ai/annadata/mqttquic/quic/QuicTypes.kt
package ai.annadata.mqttquic.quic

/**
 * QUIC stream: one bidirectional stream per MQTT connection.
 * Phase 2: ngtcp2 client + single stream.
 */
interface QuicStream {
    val streamId: Long
    suspend fun read(maxBytes: Int): ByteArray
    suspend fun write(data: ByteArray)
    suspend fun close()
}

/**
 * QUIC client: connect, TLS handshake, open one bidirectional stream.
 */
interface QuicClient {
    suspend fun connect(host: String, port: Int, connectAddress: String? = null)
    suspend fun openStream(): QuicStream
    suspend fun close()
}
